talk/owt/sdk/include/cpp/owt/base/connectionstats.h

```diff
@@ -4,13 +4,950 @@
+  static const char* const kOpen;
+  static const char* const kClosing;
+  static const char* const kClosed;
+};
+
+// https://w3c.github.io/webrtc-stats/#dom-rtcstatsicecandidatepairstate
+struct RTCStatsIceCandidatePairState {
+  static const char* const kFrozen;
+  static const char* const kWaiting;
+  static const char* const kInProgress;
+  static const char* const kFailed;
+  static const char* const kSucceeded;
+};
+
+// https://w3c.github.io/webrtc-pc/#rtcicecandidatetype-enum
+struct RTCIceCandidateType {
+  static const char* const kHost;
+  static const char* const kSrflx;
+  static const char* const kPrflx;
+  static const char* const kRelay;
+};
+
+// https://w3c.github.io/webrtc-pc/#idl-def-rtcdtlstransportstate
+struct RTCDtlsTransportState {
+  static const char* const kNew;
+  static const char* const kConnecting;
+  static const char* const kConnected;
+  static const char* const kClosed;
+  static const char* const kFailed;
+};
+
+// |RTCMediaStreamTrackStats::kind| is not an enum in the spec but the only
+// valid values are "audio" and "video".
+// https://w3c.github.io/webrtc-stats/#dom-rtcmediastreamtrackstats-kind
+struct RTCMediaStreamTrackKind {
+  static const char* const kAudio;
+  static const char* const kVideo;
+};
+
+// https://w3c.github.io/webrtc-stats/#dom-rtcnetworktype
+struct RTCNetworkType {
+  static const char* const kBluetooth;
+  static const char* const kCellular;
+  static const char* const kEthernet;
+  static const char* const kWifi;
+  static const char* const kWimax;
+  static const char* const kVpn;
+  static const char* const kUnknown;
+};
+
+// https://w3c.github.io/webrtc-stats/#dom-rtcqualitylimitationreason
+struct RTCQualityLimitationReason {
+  static const char* const kNone;
+  static const char* const kCpu;
+  static const char* const kBandwidth;
+  static const char* const kOther;
+};
+
+// https://webrtc.org/experiments/rtp-hdrext/video-content-type/
+struct RTCContentType {
+  static const char* const kUnspecified;
+  static const char* const kScreenshare;
+};
+
+struct RTCStatsType {
+  static const char* const kCodec;
+  static const char* const kInboundRTP;
+  static const char* const kOutboundRTP;
+  static const char* const kRemoteInboundRTP;
+  static const char* const kRemoteOutboundRTP;
+  static const char* const kMediaSource;
+  static const char* const kCsrc;
+  static const char* const kPeerConnection;
+  static const char* const kDataChannel;
+  static const char* const kStream;
+  static const char* const kTrack;
+  static const char* const kTranseiver;
+  static const char* const kSender;
+  static const char* const kReceiver;
+  static const char* const kTransport;
+  static const char* const kSctpTransport;
+  static const char* const kCandidatePair;
+  static const char* const kLocalCandidate;
+  static const char* const kRemoteCandidate;
+  static const char* const kCertificate;
+  static const char* const kIceServer;
+};
+
+class RTCStats {
```

Comment:
We usually don't use RTC* for OWT defined classes since they're more likely to be defined by WebRTC. But it's fine if you decide never expose WebRTC APIs to application. Otherwise, it may be confused in the future.

If you keep using RTC*, please make sure they are strictly comply with standard.

Reply:
For C++ I think we will not directly expose WebRTC APIs, and will only expose them in owt::base namespace. It will be a direct mapping from standard.